tools/acorn-optimizer.js

```diff
@@ -459,6 +459,9 @@ function isAsmUse(node) {
          ((node.object.type === 'Identifier' && // asm['X']
            node.object.name === 'asm' &&
            node.property.type === 'Literal') ||
+          (node.object.type === 'Identifier' && // exports['X']
+           node.object.name === 'exports' &&
```

Comment:
this worries me. the `exports` object is special only in the minimal runtime, and this code isn't aware of which runtime it is, so if we have a global `exports` in the normal runtime there could be confusion.

this seems like a separate issue, though? export name minification should already work with the minimal runtime, I don't see why this PR should require changes there - it's just another export?

Reply:
Its actually this line in the normal (not minimal) preable that this is needed for:  `wasmTable = exports['__indirect_function_table'];`

I want this to be detects as an asm use so that it can be correctly minified.  This needs to happend before the assignment of `Module['asm'] = exports;` because instrumentWasmExportsWithAbort actually uses the `wasmTable` globals.

Reply:
Hmm, `exports` isn't a global variable in the normal runtime AFAIK, so this would be depending on some internal detail of a function.

If this is just for metadce, then it may be best to hard-code a root for the table in `do_metadce()` in the python logic? That would still be a code size win, and we can look into optimizing it later.

Reply:
OK that makes sense I'll try that.

Reply:
Reverted this part.